drivers/interrupt_controller/intc_gicv3.c

```diff
@@ -249,14 +253,18 @@ static void gicv3_dist_init(void)
 		sys_write32(BIT_MASK(GIC_NUM_INTR_PER_REG),
 			    ICPENDR(base, idx));
 		/* All SPIs are G1S and owned by Zephyr */
-		sys_write32(0, IGROUPR(base, idx));
+		sys_write32(~0, IGROUPR(base, idx));
 		sys_write32(BIT_MASK(GIC_NUM_INTR_PER_REG),
 			    IGROUPMODR(base, idx));
 
 	}
 	/* wait for rwp on GICD */
 	gic_wait_rwp(GIC_SPI_INT_BASE);
 
+	/* Enable distributor with ARE*/
+	sys_write32(BIT(GICD_CTRL_ARE_NS) | BIT(GICD_CTLR_ENABLE_G1NS) |
+		    BIT(GICD_CTLR_ENABLE_G0), GICD_CTLR);
+
```

Comment:
Prevailing guidance on GICv3 hardware integration is to tie ARE  'HIGH' as legacy mode is deprecated feature.
But no harm in configuring explicitly in case some SOC implementation still provides ARE configuration.

Reply:
Instead of updating GICD_CTLR here you can update at line 281  G1NS/G1S based on current security state.

Reply:
'GICD_CTLR_ENABLE_G0'  can not take that to EL1NS.